vpd_tool.cpp

```diff
@@ -107,12 +107,18 @@ int main(int argc, char** argv)
 
     try
     {
-        if (objectPath.empty())
+        if ((*writeFlag || *readFlag || *dumpObjFlag) && *object &&
+            objectPath.empty())
         {
             throw runtime_error("Given path is empty.");
         }
 
-        if ((*kw) && (keyword.size() != 2))
+        if ((*writeFlag || *readFlag) && *record && (recordName.size() != 4))
+        {
+            throw runtime_error("Record " + recordName + " not supported.");
+        }
+
+        if ((*writeFlag || *readFlag) && *kw && (keyword.size() != 2))
```

Comment:
same as above command , please do not change the old working piece

Reply:
updated